### PR TITLE
cmake: Build tiflash-proxy in cmake binary dir

### LINKED ISSUE #5985: cmake --install is not working for debug profile
## Enhancement

Currently cmake always cp tiflash-proxy from `target/release`, which does not work in debug profile.

### PR DESCRIPTION
**Pull request by breezewish**
Signed-off-by: Wish <breezewish@outlook.com>

### What problem does this PR solve?

Issue Number: close #5985

### What is changed and how it works?

1. Deprecate and remove the CMake option `USE_INTERNAL_TIFLASH_PROXY`. If we want to work with a specific TiFlash proxy branch, we could just cd into `contrib/tiflash-proxy` and develop it as a normal git repository. There is simply no need for specifying a directory containing an external TiFlash proxy.

2. Change `--compress-debug-sections=zlib-gnu` to `--compress-debug-sections`, as `zlib-gnu` is not available any more starting from LLVM 15. See https://github.com/llvm/llvm-project/commit/45ae553109af8e1e71ae5ddf3d92fc1bf531aea1

3. Build TiFlash proxy in the CMake directory (e.g., `cmake-build-debug`). Previously TiFlash proxy will be built in `tiflash/contrib/tiflash-proxy/target/...`. Now it will be built in `cmake-build-debug/contrib/tiflash-proxy-cmake/...`. This makes it identical with `libsymbolization` and allows us to clean build directories easier. Also, when we want to cache compile artifacts, we only need to deal with one directory.

   - [x] TiFlash proxy hard coded the build path. https://github.com/pingcap/tidb-engine-ext/pull/186 fixed it. We should merge that PR first.

4. After building, RPATH for TiFlash proxy is set to a relative path, instead of an absolute path. In this way, when the build directory is moved to another place, the TiFlash proxy resolution will be still fine.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -230,16 +230,6 @@ cmake .. -GNinja -DCMAKE_BUILD_TYPE=DEBUG -DFOO=BAR
 
   Specifically, for [TiFlash proxy](https://github.com/pingcap/tidb-engine-ext):
```
Comment (Lloyd-Pottiger) [inline README.md:231]:
```suggestion
```

seems can be removed?

**Review by Lloyd-Pottiger — APPROVED** [review]
Others LGTM

**Review by JaySon-Huang — APPROVED** [review]

**Comment by JaySon-Huang** [comment]
Will "Build TiFlash proxy in the CMake directory" make the ProxyCache always invalid and slow down the build time when proxy is not changed?

https://github.com/PingCAP-QE/ci/blob/main/jenkins/pipelines/ci/tiflash/tiflash-build-common.groovy#L305

**Comment by JaySon-Huang** [comment]
> Will "Build TiFlash proxy in the CMake directory" make the ProxyCache always invalid and slow down the build time when proxy is not changed?
> 
> https://github.com/PingCAP-QE/ci/blob/main/jenkins/pipelines/ci/tiflash/tiflash-build-common.groovy#L305

This PR will make the `build` always build proxy, even if the proxy code is not changed. The build time is slightly increase about 6 min

```
[2022-10-12T10:55:45.309Z] proxy cache found
...
[2022-10-12T11:04:21.633Z]    Compiling raftstore-proxy v0.0.1 (/home/jenkins/agent/workspace/tiflash-build-common/tiflash/contrib/tiflash-proxy/raftstore-proxy)
[2022-10-12T11:04:26.881Z]     Finished dev [unoptimized] target(s) in 5m 20s
```

**Comment by breezewish** [comment]
> Will "Build TiFlash proxy in the CMake directory" make the ProxyCache always invalid and slow down the build time when proxy is not changed?
> 
> https://github.com/PingCAP-QE/ci/blob/main/jenkins/pipelines/ci/tiflash/tiflash-build-common.groovy#L305

1. If you keep the cmake build directory, and no rust source is modified, Rust building is completely skipped.
2. If you keep the cmake build directory, and some rust source is modified, Rust will utilize current build cache and build the changed part.

However, if the cmake build directory is dropped, then Proxy need to be compiled from beginning.

**Comment by breezewish** [comment]
> > Will "Build TiFlash proxy in the CMake directory" make the ProxyCache always invalid and slow down the build time when proxy is not changed?
> > 
> > https://github.com/PingCAP-QE/ci/blob/main/jenkins/pipelines/ci/tiflash/tiflash-build-common.groovy#L305
> 
> This PR will make the `build` always build proxy, even if the proxy code is not changed. The build time is slightly increase about 6 min
> 
> https://ci.pingcap.net/blue/organizations/jenkins/tiflash-build-common/detail/tiflash-build-common/14112/pipeline/177
> 
> ```
> [2022-10-12T10:55:45.309Z] proxy cache found
> ...
> [2022-10-12T11:04:21.633Z]    Compiling raftstore-proxy v0.0.1 (/home/jenkins/agent/workspace/tiflash-build-common/tiflash/contrib/tiflash-proxy/raftstore-proxy)
> [2022-10-12T11:04:26.881Z]     Finished dev [unoptimized] target(s) in 5m 20s
> ```

Interesting, let me check out why CI does not skip the Rust part when no source code is changed.

**Pull request merged**